crates/stdarch-gen/neon.spec

```diff
@@ -1955,6 +2069,143 @@ validate 9, 30, 11, 20, 13, 18, 15, 48
 aarch64 = pmull
 generate poly8x16_t:poly8x16_t:poly16x8_t
 
+/// Polynomial multiply long
+name = vmull_high
+no-q
+multi_fn = vmull-noqself-noext, {simd_extract, a, 1}, {simd_extract, b, 1}
+a = 1, 15
+b = 1, 3
+validate 17
+
+aarch64 = pmull2
```

Comment:
`pmull` with the `p128` type should depend on the `crypto` feature.

Reply:
While you're at it, could you move `vmull_p64` into the code generator as well? It should be available on both arm and aarch64 with the `crypto` feature enabled.

Reply:
So should we use `#[target_feature(enable = "neon, crypto")]`?

Reply:
Yes. Although keep in mind that in the future we will split the `crypto` feature into separate `aes` and `sha2` features to match LLVM.

Reply:
Ok. But aes and sha2feature do not seem to affect the implementation of vmull

Reply:
I did not find the implementation of vmull_p64 on arm in llvm, so I don’t know what link should be used

Reply:
I checked Clang and it seems to be missing this intrinsic on ARM. However I checked the LLVM source code and you should be able to make it generate the correct instruction with `llvm.arm.neon.vmullp.v2i64`. However I have not tested it so it may not work.

Reply:
It looks like it cannot compile successfully:

```
#[cfg(target_arch = "arm")]
#[target_feature(enable = "neon,crypto,v8")]
pub unsafe fn vmull_p64(a: p64, b: p64) -> p128 {
    #[allow(improper_ctypes)]
    extern "C" {
        #[cfg_attr(target_arch = "arm", link_name = "llvm.arm.neon.vmullp.v2i64")]
        fn vmull_p64_(a: int64x1_t, b: int64x1_t) -> int64x2_t;
    }
transmute(vmull_p64_(transmute(a), transmute(b)))
}
```
```
Intrinsic has incorrect argument type!
<2 x i64> (<1 x i64>, <1 x i64>)* @llvm.arm.neon.vmullp.v2i64
in function _ZN7example9vmull_p6417hc3e4f087b1fb350cE
LLVM ERROR: Broken function found, compilation aborted!
```
I tried to find a solution in the source code of llvm, but it seems that it should use v2i64. I don't know where the problem is
```
  def  VMULLp8   :  N3VLInt<0, 1, 0b00, 0b1110, 0, IIC_VMULi16D, "vmull", "p8",
                            v8i16, v8i8, int_arm_neon_vmullp, 1>;
  def  VMULLp64  : N3VLIntnp<0b00101, 0b10, 0b1110, 0, 0, NoItinerary,
                          "vmull", "p64", v2i64, v1i64, int_arm_neon_vmullp, 1>,
                    Requires<[HasV8, HasCrypto]>;
```